fe1-web/src/features/linked-organizations/network/LinkedOrgHandler.ts

```diff
@@ -0,0 +1,150 @@
+import { ActionType, ObjectType, ProcessableMessage } from 'core/network/jsonrpc/messages';
+import { dispatch } from 'core/redux';
+
+import { LinkedOrganizationsConfiguration } from '../interface';
+import { Challenge } from '../objects/Challenge';
+import { addChallenge } from '../reducer';
+import { ChallengeRequest } from './messages';
+import { ChallengeMessage } from './messages/ChallengeMessage';
+import { FederationExpect } from './messages/FederationExpect';
+import { FederationInit } from './messages/FederationInit';
+
+/**
+ * Handler for linked organization messages
+ */
+
+/**
+ * Handles an ChallengeRequest message.
+ */
+export const handleChallengeMessage = () => (msg: ProcessableMessage) => {
+  if (
+    msg.messageData.object !== ObjectType.FEDERATION ||
+    msg.messageData.action !== ActionType.CHALLENGE
+  ) {
+    console.warn('handleRequestChallengeMessage was called to process an unsupported message');
+    return false;
+  }
+
+  const makeErr = (err: string) => `challenge was not processed: ${err}`;
+
+  // obtain the lao id from the channel
+  if (!msg.laoId) {
+    console.warn(makeErr('message was not sent on a lao subchannel'));
+    return false;
+  }
+  try {
+    const challengeMessage = msg.messageData as ChallengeMessage;
+
+    const jsonObj = {
+      value: challengeMessage.value.toString(),
+      valid_until: challengeMessage.valid_until,
+    };
+    const challenge = Challenge.fromJson(jsonObj);
+    dispatch(addChallenge(msg.laoId, challenge.toState()));
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+/**
+ * Handles an requestChallenge message.
+ */
+export const handleChallengeRequestMessage =
+  (getCurrentLaoId: LinkedOrganizationsConfiguration['getCurrentLaoId']) =>
+  (msg: ProcessableMessage) => {
+    if (
+      msg.messageData.object !== ObjectType.FEDERATION ||
+      msg.messageData.action !== ActionType.CHALLENGE_REQUEST
+    ) {
+      console.warn('handleRequestChallengeMessage was called to process an unsupported message');
+      return false;
+    }
+
+    const makeErr = (err: string) => `challenge/request was not processed: ${err}`;
+
+    const laoId = getCurrentLaoId();
+    if (!laoId) {
+      console.warn(makeErr('no Lao is currently active'));
+      return false;
+    }
+    if (msg.messageData instanceof ChallengeRequest) {
```

Comment:
Given we don't do this check in any other handler, why did you add it here? Could we add this to all other handlers instead / on top of the `as X` ?

Reply:
I think this also has something to do with that I don't save the ChallengeRequest. Most of the other handlers do the ... as XMessage and then do a try catch to create the Actual Object (e.g. RollCall) out of it. Since I am not creating an Object of the message I just checked it with the instanceOf and then checked for my variables (e.g. timestamp)